tests/ui/traits/new-solver/equating-projection-cyclically.rs

```diff
@@ -1,5 +1,7 @@
+// check-pass
 // compile-flags: -Ztrait-solver=next
-// known-bug: unknown
+
+// FIXME(-Ztrait-solver=next): what is this testing and why did we add this test.
```

Comment:
This was supposed to be testing `<?0 as Trait>::Assoc = ?0` hitting the occurs check.

Reply:
yeah, broke because of variance for some reason :sweat_smile: change it to force the infer var to be invariant